From 9b09f8faa51e6c6c4fdc90933666f6eeee4fe386 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Mon, 22 Sep 2025 18:06:51 -0600
Subject: [PATCH] FIX: Remove overly strict CutTools compiler requirements

* Don't force the compiler in use in the conda environment to be the
  exact same compiler used to compile the vendored CutTools.
---
 madgraph/interface/amcatnlo_run_interface.py | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/madgraph/interface/amcatnlo_run_interface.py b/madgraph/interface/amcatnlo_run_interface.py
index 38a4605..fd08010 100755
--- a/madgraph/interface/amcatnlo_run_interface.py
+++ b/madgraph/interface/amcatnlo_run_interface.py
@@ -5448,26 +5448,6 @@ PYTHIA8LINKLIBS=%(pythia8_prefix)s/lib/libpythia8.a -lz -ldl"""%{'pythia8_prefix
             not os.path.exists(os.path.realpath(pjoin(libdir, 'mpmodule.mod'))):
             raise aMCatNLOError('CutTools compilation failed.')            
 
-        # Verify compatibility between current compiler and the one which was
-        # used when last compiling CutTools (if specified).
-        compiler_log_path = pjoin(os.path.dirname((os.path.realpath(pjoin(
-                                  libdir, 'libcts.a')))),'compiler_version.log')
-        if os.path.exists(compiler_log_path):
-            compiler_version_used = open(compiler_log_path,'r').read()
-            if not str(misc.get_gfortran_version(misc.detect_current_compiler(\
-                       pjoin(sourcedir,'make_opts')))) in compiler_version_used:
-                if os.path.exists(pjoin(sourcedir,'CutTools')):
-                    logger.info('CutTools was compiled with a different fortran'+\
-                                            ' compiler. Re-compiling it now...')
-                    misc.compile(['cleanCT'], cwd = sourcedir)
-                    misc.compile(['CutTools','-j1'], cwd = sourcedir, nb_core=1)
-                    logger.info('          ...done.')
-                else:
-                    raise aMCatNLOError("CutTools installation in %s"\
-                                 %os.path.realpath(pjoin(libdir, 'libcts.a'))+\
-                 " seems to have been compiled with a different compiler than"+\
-                    " the one specified in MG5_aMC. Please recompile CutTools.")
-
         # make IREGI (only necessary with MG option output_dependencies='internal')
         if not os.path.exists(os.path.realpath(pjoin(libdir, 'libiregi.a'))) \
            and os.path.exists(pjoin(sourcedir,'IREGI')):
-- 
2.51.0

